contracts/src/instances/savings-usds-l2/SavingsUSDSL2Base.sol

```diff
@@ -0,0 +1,176 @@
+
+    IPSM internal immutable _PSM;
+
+    /// @notice instantiates the SavingsUSDSL2Base contract.
+    /// @param __PSM The PSM contract.
+    constructor(IPSM __PSM) {
+        // Initialize the PSM immutable.
+        _PSM = __PSM;
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a deposit from the user in base.
+    /// @param _baseAmount The base amount to deposit.
+    /// @return The shares that were minted in the deposit.
+    /// @return The amount of ETH to refund. Since this yield source isn't
+    ///         payable, this is always zero.
+    function _depositWithBase(
+        uint256 _baseAmount,
+        bytes calldata // unused
+    ) internal override returns (uint256, uint256) {
+        if (_baseAmount == 0) {
+            // The PSM swap function will revert if the amount is 0,
+            // but the output would just be 0 anyway.
+            return (0, 0);
+        }
+
+        // Take custody of the deposit in base.
+        ERC20(address(_baseToken)).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _baseAmount
+        );
+
+        // Deposit the base into the yield source.
+        //
+        // NOTE: We increase the required approval amount by 1 wei so that
+        // the vault ends with an approval of 1 wei. This makes future
+        // approvals cheaper by keeping the storage slot warm.
+        ERC20(address(_baseToken)).forceApprove(address(_PSM), _baseAmount + 1);
+
+        // Depositing amounts to swapping USDS for SUSDS in the PSM.
+        uint256 sharesMinted = _PSM.swapExactIn(
+            address(_baseToken),
+            address(_vaultSharesToken),
+            _baseAmount,
+            _convertToShares(_baseAmount) - 1,
+            address(this),
+            0
+        );
+
+        return (sharesMinted, 0);
+    }
+
+    /// @dev Process a deposit in vault shares.
+    /// @param _shareAmount The vault shares amount to deposit.
+    function _depositWithShares(
+        uint256 _shareAmount,
+        bytes calldata // unused _extraData
+    ) internal override {
+        ERC20(address(_vaultSharesToken)).safeTransferFrom(
+            msg.sender,
+            address(this),
+            _shareAmount
+        );
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    /// @return amountWithdrawn The amount of base withdrawn.
+    function _withdrawWithBase(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override returns (uint256 amountWithdrawn) {
+        if (_shareAmount <= 1) {
+            // The PSM swap function will revert if the amount to swap is 0,
+            // but the output would just be 0 anyway.
+            return 0;
+        }
+        ERC20(address(_vaultSharesToken)).forceApprove(
```

Comment:
Let’s either comment this or move it under the next comment block to follow the style guide

Reply:
Done!